src/vm/methodtable.cpp

```diff
@@ -2458,6 +2465,67 @@ bool MethodTable::ClassifyEightBytesWithManagedLayout(SystemVStructRegisterPassi
             continue;
         }
 
+        if (fieldClassificationType == SystemVClassificationTypeTypedReference || 
+            CorInfoType2UnixAmd64Classification(GetClass_NoLogging()->GetInternalCorElementType()) == SystemVClassificationTypeTypedReference)
+        {
+            // The TypedReference is a very special type.
```

Comment:
The classifier is reporting wrong classification for the TypedReference type. In source the type is IntPtr, but when the VM instantiates the type it is making the first eightbyte of type ByRef. The classifier needs to report the correct type of the two eightbytes for the type TypedReference. Once this is done, there is no need to change anything in the Jit. The morper and all the other phases (including handling the returns for the TypedReference will work properly). This change is not needed by the Jit only. The classification is also used internally by the VM when it packs parameters for reflection calls.